Tdl 17006 revert back companies to incremental

savan-chovatiya opened the pull request:
# Description of change
[TDl-17006](https://jira.talendforge.org/browse/TDL-17006): Revert back the companies stream to work as an INCREMENTAL stream
- Reverted back companies stream to incremental again same as v1.1.3.

**NOTE**: Bookmark integration test was failing for companies after changing it to incremental. It got the same records for both syncs in the bookmark test but we don't have a token used in CIrcleCI to validate possible dates to check that test.
For passing build, added _companies_ to untestable_stream and created a backlog card [TDL-17035](https://jira.talendforge.org/browse/TDL-17035) for the fix. ([Slack Thread](https://talend.slack.com/archives/C01TGC4QAUE/p1640870530022000))

# Manual QA steps
 - Verified that companies stream is marked as incremental in catalog with valid replication key and inclusion.
- Verified that a number of records differ as per start_date/bookmark and also respect bookmark/start_date.

 
# Risks
 - 
 
# Rollback steps
 - revert this branch

savan-chovatiya commented on tap_intercom/sync.py:
```diff
@@ -35,9 +35,7 @@ def translate_state(state):
-            # Stream `companies` is changed from incremental to full_table
-            # so adding replication key used at incremental time to keep consistency.
-            replication_key = 'updated_at' if stream == "companies" else STREAMS[stream].replication_key
+            replication_key = STREAMS[stream].replication_key
```
The companies stream is changed to incremental again so it has replication_key now so no need to handle it specifically.

dmosorast reviewed (changes requested):
I looks like the integration test is failing becuase the tap isn't using the bookmark on the second sync.

```
======================================================================
FAIL: test_run (test_intercom_bookmarks.IntercomBookmarks) (stream='companies')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/project/tests/test_intercom_bookmarks.py", line 187, in test_run
    self.assertLess(second_sync_count, first_sync_count)
AssertionError: 162 not less than 162
```

savan-chovatiya commented:
> For passing build, added _companies_ to untestable_stream and created a backlog card [TDL-17035](https://jira.talendforge.org/browse/TDL-17035) for the fix. ([Slack Thread](https://talend.slack.com/archives/C01TGC4QAUE/p1640870530022000))

For passing build, added _companies_ to untestable stream and created backlog card to fix it as per discussion in this [Slack Thread](https://talend.slack.com/archives/C01TGC4QAUE/p1640870530022000).

savan-chovatiya commented:
> I looks like the integration test is failing becuase the tap isn't using the bookmark on the second sync.
> 
> ```
> ======================================================================
> FAIL: test_run (test_intercom_bookmarks.IntercomBookmarks) (stream='companies')
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/root/project/tests/test_intercom_bookmarks.py", line 187, in test_run
>     self.assertLess(second_sync_count, first_sync_count)
> AssertionError: 162 not less than 162
> ```

For passing build, added _companies_ to untestable stream and created backlog card to fix it as per discussion in this [Slack Thread](https://talend.slack.com/archives/C01TGC4QAUE/p1640870530022000).

dmosorast reviewed (approved):
Discussed offline, the test is manufacturing bookmarks, so the test failure is not necessarily a bug. Capturing the work, and functional testing should be sufficient. Approved ✅
